stackslib/src/chainstate/nakamoto/tenure.rs

```diff
@@ -595,6 +595,53 @@ impl NakamotoChainState {
         Ok(tenure_opt)
     }
 
+    /// Get the tenure change that was active for a given block header
+    /// If a tenure change occurred during this block, it will be returned
+    pub fn get_tenure_for_block(
```

Comment:
This method is giving me lots of pause.  Each Nakamoto block header contains the tenure ID consensus hash.  Why can't we load the tenure using that -- i.e. by matching against `tenure_id_consensus_hash`?

Reply:
I need the most recent TenureChange payload, _not_ the `TenureChangeCause::BlockFound`. Maybe my mental model of the tenure change payloads is wrong, but my understanding is:

```
block.header.consensus_hash := identifies the bitcoin block during which the miner of `block` was selected 
tenure_id_consensus_hash := identifies the bitcoin block during which the miner of `block` was selected
burn_view_consensus_hash := identifies the bitcoin block that subsequent blocks should have as their "view"
```

Taken together, this means that if there's a sortition in bitcoin block 1, and not one in bitcoin block 2, we'd expects nakamoto blocks like:

```
Block A: TenureChange with tenure_id = 1, burn_view = 1.
Block B: TenureChange with tenure_id = 1, burn_view = 2.
Block C: No tenure change. tenure_id = 1, but burn_view = 2.
Block D: No tenure change. tenure_id = 1, but burn_view = 2.
...
```

So, in order to determine the burn view of _Block D_, we need to find the closest tenure change payload. This method does that.